architecture/wm_sca/Readme.md

```diff
@@ -0,0 +1,36 @@
+<!---
+Copyright (C) 2015-2021, Wazuh Inc.
+Created by Wazuh, Inc. <info@wazuh.com>.
+This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+-->
+
+# Wazuh module: SCA architecture
+## Index
+1. [Purpose](#purpose)
+2. [Sequence Diagram](#sequence-diagram)
+
+## Purpose
+The security configuration assessment module (**SCA**) performs hardening and configuration scans based on custom policies, created and maintained by Wazuh's team based on CIS benchmarks.
+
+This module is composed of a main thread, from which two other secondary threads diverge. These threads are:
+
+- The main thread: In charge of parsing and converting to **JSON** the **YML** policies. These policies are then stored and executed accordingly. This thread contains high amounts of memory management and should be looked at closely, especially the `cis_db_for_hash` global structure, which seems to contain detailed information from all checks.
+
+- `wm_sca_dump_db_thread`: In charge of dumping the scan results into the manager's **SCA** database.
+
+- `wm_sca_request_thread`: In charge of requesting information if missing data is detected. Non-suspicious memory management.
+
+
+## Sequence Diagram
+The provided sequence diagram shows the basic flow of Wazuh's **SCA** module. The main steps are:
+1. **SCA**'s main thread starts working according to the configuration provided in `ossec.conf`.
+2. Policies are stored in memory.
```

Comment:
1. SCA start-up process:
  1.1. Load configuration
  1.2. Structure initialization
  1.3. Threads `wm_sca_dump_db_thread` and `wm_sca_request_thread` are created
2. The main thread is in charge of running the scans every interval:
  2.1. The SCA policy is dumped in memory
  2.2. The scan is performed
  2.3. Results are stored in memory
3. Once the results are available, they are synchronized with the manager (`wm_sca_dump_db_thread`)
4. The `wm_sca_request_thread` keeps processing manager requests to ensure the synchronization succeeded